src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2936,10 +2946,9 @@ private OctetsFW encodeSections(
             OctetsFW payload)
         {
             AmqpSectionEncoder previous = null;
-            DirectBuffer buffer = payload.buffer();
-            int offset = payload.offset();
-            int progress = offset;
-            int limit = payload.limit();
+            DirectBuffer buffer = payload != null ? payload.buffer() : null;
+            int progress = payload != null ? payload.offset() : -1;
+            int limit = payload != null ? payload.limit() : -1;
```

Comment:
Suggest not attempting to handle `null` case in generic codepath.
Instead put `assert payload != null` here, and then in `encodeFragmentInit` above, consider changing
```
            encodeMessageProperties(dataEx.properties());
            encodeMessageAnnotations(dataEx.annotations());
            encodeApplicationProperties(dataEx.applicationProperties());
            this.sectionEncoder = lookupBodyEncoder(bodyKind);
            return encodeSections(deferred, payload);
```
to
```
            encodeMessageProperties(dataEx.properties());
            encodeMessageAnnotations(dataEx.annotations());
            encodeApplicationProperties(dataEx.applicationProperties());
            if (payload == null)
            {
                return encodeSectionValueNull(deferred, payload);
            }
            else
            {
                this.sectionEncoder = lookupBodyEncoder(bodyKind);
                return encodeSections(deferred, payload);
            }
```
You could then inline `encodeSectionValueNull` instead, and return a pre-combined buffer with `VALUE` descriptor followed by `NULL` value rather than copying them into a combined buffer each time.